examples/contrib/oed/eig_estimation_benchmarking.py

```diff
@@ -7,148 +9,383 @@
-     [(X_lm, lm_true_eig, [torch.tensor([10., .1])]),
-      (X_lm, vi_for_lm, [torch.tensor([10., .1]), 10000, 1]),
-      (X_lm, naive_rainforth_lm, [torch.tensor([10., .1]), 2000, 2000]),
-      (X_lm, donsker_varadhan_lm, [torch.tensor([10., .1]), 4000, 200, 0.005, T_neural(2, 2)])],
+        return torch.tensor([0.5*torch.logdet(2*np.pi*np.e*C) for C in target_posterior_covs])
+
+
+def lm_H_prior(model, design, observation_labels, target_labels):
+    if isinstance(target_labels, str):
+        target_labels = [target_labels]
+
+    w_sd = torch.cat(list(model.w_sds.values()), dim=-1)
+    prior_cov = torch.diag(w_sd**2)
+    target_indices = get_indices(target_labels, tensors=model.w_sds)
+    target_prior_covs = prior_cov[target_indices, :][:, target_indices]
+    return 0.5*torch.logdet(2*np.pi*np.e*target_prior_covs)
+
+
+def mc_H_prior(model, design, observation_labels, target_labels, num_samples=1000):
+    if isinstance(target_labels, str):
+        target_labels = [target_labels]
+
+    expanded_design = lexpand(design, num_samples)
+    trace = pyro.poutine.trace(model).get_trace(expanded_design)
+    trace.compute_log_prob()
+    lp = sum(trace.nodes[l]["log_prob"] for l in target_labels)
+    return -lp.sum(0)/num_samples
+
+
+def vi_eig_lm(model, design, observation_labels, target_labels, *args, **kwargs):
+    # **Only** applies to linear models - analytic prior entropy
+    ape = vi_ape(model, design, observation_labels, target_labels, *args, **kwargs)
+    prior_entropy = lm_H_prior(model, design, observation_labels, target_labels)
+    return prior_entropy - ape
+
+
+def ba_eig_lm(model, design, observation_labels, target_labels, *args, **kwargs):
+    # **Only** applies to linear models - analytic prior entropy
+    ape = barber_agakov_ape(model, design, observation_labels, target_labels, *args, **kwargs)
+    prior_entropy = lm_H_prior(model, design, observation_labels, target_labels)
+    return prior_entropy - ape
+
+
+def ba_eig_mc(model, design, observation_labels, target_labels, *args, **kwargs):
+    # Compute the prior entropy my Monte Carlo, the uses barber_agakov_ape
+    if "num_hprior_samples" in kwargs:
+        hprior = mc_H_prior(model, design, observation_labels, target_labels, kwargs["num_hprior_samples"])
+    else:
+        hprior = mc_H_prior(model, design, observation_labels, target_labels)
+    return hprior - barber_agakov_ape(model, design, observation_labels, target_labels, *args, **kwargs)
+
+
+# Makes the plots look pretty
+vi_eig_lm.name = "Variational inference"
+vi_ape.name = "Variational inference"
+ba_eig_lm.name = "Barber-Agakov"
+ba_eig_mc.name = "Barber-Agakov"
+barber_agakov_ape.name = "Barber-Agakov"
+donsker_varadhan_eig.name = "Donsker-Varadhan"
+linear_model_ground_truth.name = "Ground truth"
+naive_rainforth_eig.name = "Naive Rainforth"
+
+
+@pytest.mark.parametrize("title,model,design,observation_label,target_label,arglist", [
+    ("Sigmoid link function",
```

Comment:
Move these test cases into a list of tuples so they're easier to read; see e.g. [HMC tests](https://github.com/uber/pyro/blob/dev/tests/infer/mcmc/test_hmc.py#L60)